frontend/src/Troop/troopReducer.js

```diff
@@ -0,0 +1,20 @@
+import { FETCH_TROOPS_SUCCESS, FETCH_TROOPS_REQUEST, FETCH_TROOPS_FAILURE } from './actionCreator';
+
+const initialState = {
+  troops: [],
+  isLoading: false,
+  error: undefined,
+};
+
+export default function troop(state = initialState, action) {
+  switch (action.type) {
+    case FETCH_TROOPS_REQUEST:
+      return { ...state, isLoading: true };
+    case FETCH_TROOPS_SUCCESS:
+      return { ...state, troops: action.payload, isLoading: false };
+    case FETCH_TROOPS_FAILURE:
+      return { ...state, error: action.payload, isLoading: false };
```

Comment:
```suggestion
      return { ...state, isLoading: false, error: action.payload };
```

please keep the order of these things from the initialState for readability
troops, isLoading, error